flow/service.go

```diff
@@ -4,43 +4,101 @@ import (
 	"context"
 	"errors"
 	"github.com/coreos/etcd/clientv3"
+	"strings"
 	"time"
 )
 
 // Service это интерфейс нашего сервиса Flow
 type Service interface {
+	FlowDelete(flowDeleteRequest) (bool, error)
+	FlowPost(flowPostRequest) (bool, error)
 	FlowPut(flowPutRequest) (bool, error)
 	FlowGet(id string) (string, error)
 }
 
-// ErrInvalidArgument is returned when one or more arguments are invalid.
-var ErrInvalidArgument = errors.New("invalid argument")
-
 type service struct {
-	etcdDataBaseClientApi clientv3.KV
+	etcdClientApi clientv3.KV
 }
 
-func NewService(etcdDataBaseClientApi clientv3.KV) Service {
-	return &service{etcdDataBaseClientApi}
+func NewService(etcdClientApi clientv3.KV) Service {
+	return &service{etcdClientApi}
 }
 
-func (f *service) FlowPut(conf flowPutRequest) (bool, error) {
+func (f *service) FlowDelete(conf flowDeleteRequest) (bool, error) {
+
+	if conf.Id == "" {
+		return false, errors.New("id is required")
+	}
+	if !strings.HasPrefix(conf.Id, "/templates/") {
+		return false, errors.New("id should start with '/template/'")
+	}
+	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
The second return value is error?
```suggestion
	ctxForEtcd, _ := context.WithTimeout(context.Background(), 10*time.Second)
```